app/components/server_icon/server_icon.tsx

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {StyleSheet, View} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import {useTheme} from '@context/theme';
+import {changeOpacity} from '@utils/theme';
+
+const styles = StyleSheet.create({
+    icon: {
+        flex: 0,
+        overflow: 'hidden',
+        alignItems: 'center',
+        justifyContent: 'center',
+        position: 'absolute',
+        zIndex: 10,
+        top: 10,
+        left: 16,
+        width: 40,
+        height: 40,
+    },
+});
+
+export default function ServerIcon() {
```

Comment:
I would want to keep this component as part of the `channel_list` screen components and not really a generic one if that is ok

Reply:
Yeah I can move it into the `components/channel_list` but I moved it out as it was being called in both the `ChannelList` and also the `TeamSidebar`.